python-sdk/nuscenes/eval/lidarseg/utils.py

```diff
@@ -0,0 +1,331 @@
+                 class index.
+        """
+        conf = self.global_cm.copy()
+
+        # Get the intersection for each class.
+        intersection = np.diagonal(conf)
+
+        # Get the union for each class.
+        ground_truth_set = conf.sum(axis=1)
+        predicted_set = conf.sum(axis=0)
+        union = ground_truth_set + predicted_set - intersection
+
+        # Get the IOU for each class.
+        # In case we get a division by 0, ignore / hide the error(adapted from
+        # https://github.com/davidtvs/PyTorch-ENet/blob/master/metric/iou.py).
+        with np.errstate(divide='ignore', invalid='ignore'):
+            iou_per_class = intersection / (union.astype(np.float32))
+
+        return iou_per_class
+
+    def get_mean_iou(self) -> float:
+        """
+        Gets the mean IOU (mIOU) over the classes.
+        :return: mIOU over the classes.
+        """
+        iou_per_class = self.get_per_class_iou()
+        miou = float(np.nanmean(iou_per_class))
+        return miou
+
+    def get_freqweighted_iou(self) -> float:
+        """
+        Gets the frequency-weighted IOU over the classes.
+        :return: Frequency-weighted IOU over the classes.
+        """
+        conf = self.global_cm.copy()
+
+        # Get the number of points per class (based on ground truth).
+        num_points_per_class = conf.sum(axis=1)
+
+        # Get the total number of points in the eval set.
+        num_points_total = conf.sum()
+
+        # Get the IOU per class.
+        iou_per_class = self.get_per_class_iou()
+
+        # Weight the IOU by frequency and sum across the classes.
+        freqweighted_iou = float(np.nansum(num_points_per_class * iou_per_class) / num_points_total)
+
+        return freqweighted_iou
+
+
+class LidarsegClassMapper:
+    """
+    Maps the (raw) classes in nuScenes-lidarseg to the (merged) classes for the nuScenes-lidarseg challenge.
+
+    Example usage::
+        nusc_ = NuScenes(version='v1.0-mini', dataroot='/data/sets/nuscenes', verbose=True)
+        mapper_ = LidarsegClassMapper(nusc_)
+    """
+    def __init__(self, nusc: NuScenes):
+        """
+        Initialize a LidarsegClassMapper object.
+        :param nusc: A NuScenes object.
+        """
+        self.nusc = nusc
+
+        self.ignore_class = self.get_ignore_class()
+
+        self.raw_name_2_merged_name_mapping = self.get_raw2merged()
+        self.merged_name_2_merged_idx_mapping = self.get_merged2idx()
+
+        self.check_mapping()
+
+        self.raw_idx_2_merged_idx_mapping = self.get_raw_idx_2_merged_idx()
+
+    @staticmethod
+    def get_ignore_class() -> Dict[str, int]:
+        """
+        Defines the name and index of the ignore class.
+        :return: A dictionary containing the name and index of the ignore class.
+        """
+        return {'name': 'ignore', 'index': 0}
+
+    def get_raw2merged(self) -> Dict:
```

Comment:
probably minor but maybe `fine2coarse_mapping` is better?

Reply:
`coarse2fine` is it? Since we are going from 32 classes to 16 classes

Reply:
Renamed all relevant variables / methods